scripts/webpack/webpack.common.ts

```diff
@@ -5,12 +5,23 @@ import fs from 'fs';
 import HtmlWebpackPlugin from 'html-webpack-plugin';
 import MiniCssExtractPlugin from 'mini-css-extract-plugin';
 import CopyPlugin from 'copy-webpack-plugin';
+import { ESBuildMinifyPlugin } from 'esbuild-loader';
 
 import { getAlias, getJsLoader, getStyleLoaders } from './shared';
 
 const pages = glob
-  .sync('./webapp/templates/*.html')
+  // Most of the cases we will be developing the SPA
+  // So it makes sense only building it
+  // We go from
+  // [webpack.Progress]  |  | 1131 ms asset processing > HtmlWebpackPlugin
+  // To  [webpack.Progress]  |  | 215 ms asset processing > HtmlWebpackPlugin
+  .sync(
+    process.env.NODE_ENV === 'production'
+      ? './webapp/templates/*.html'
+      : './webapp/templates/index.html'
+  )
```

Comment:
process.env.NODE_ENV isn't really reliable, we could use
https://webpack.js.org/configuration/mode/#mode-none instead

Reply:
Does this mean in dev mode you won't be able to test other pages? I understand the rationale, but I think this might be confusing when we do want to edit those pages.

It looks like the only thing we do with `HtmlWebpackPlugin` is this `<%= webpack.hash %>` Interpolation, and 215ms to do some string replacements is still very very slow imo.

Maybe we could git rid of HtmlWebpackPlugin altogether and instead export `webpack.hash` to some sort of a file, and then on the go side we would read this file each time we need to render a page and do interpolation via go templates?

I'm not sure how this would work with things like livereload, though.

@eh-am What do you think?

Reply:
I guess I just remembered we also wanted to get rid of templating on the go side, let's maybe discuss this part on a call.

Reply:
>but I think this might be confusing when we do want to edit those pages.
100% agree, I hope to get rid of this "optimization" soon.

>and 215ms to do some string replacements is still very very slow imo.
I read on the [webpack docs](https://webpack.js.org/guides/build-performance/#avoid-production-specific-tooling) that stuff like hash isn't very good for production. I didn't profile but I trust them (actually I did but couldn't pin point to the exact problem). I guess we can just not hash for dev? Unless HMR use it for refreshing somehow, but we don't have it so it's a problem for the future.